lucene/facet/src/java/org/apache/lucene/facet/FacetsConfig.java

```diff
@@ -372,7 +403,10 @@ private void processFacetFields(
 
         // Drill down:
```

Comment:
Hmm, maybe this is just a personal preference, but I'd find it more readable to do:

```suggestion
        // Drill down:
        if (disableDrillDown == false) {
          int start;
          if (ft.requireDimensionDrillDown) {
            start = 1;
          } else {
            start = 2;
          }
          for (int i = start; i <= cp.length; i++) {
            doc.add(new StringField(indexFieldName, pathToString(cp.components, i), Field.Store.NO));
          }
        }
```

Reply:
I think this logic may or may not be correct actually, as it depends on whether full path terms is considered as "drill down terms" (see my question to Mike above).

Also, the index `i` in `pathToString(cp.components, i)` above is used as "length". So with the original implementation when `start = cp.length`,  `pathToString(cp.components, start)`  will still be executed and give the full path including every element inside `cp.components`

Reply:
I think this approach is good, i.e. "disable drill down" would also mean "disable the full path drill down", i.e. it would mean this `dim` adds zero tokens to the inverted index.

Reply:
Updated to use the new `DrillDownTermsIndexing.NONE` option to denote the all drill down disabled condition. Also refactored a bit more there to support the `DrillDownTermsIndexing.DIMENSION_AND_FULL_PATH` option that was previously commented out.